frontend/src/main/components/Ride/RideForm.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import { Button, Form } from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom';
+function RideForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+
+    
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker enable all
+    const navigate = useNavigate();
+
+    //const testIdPrefix = "RideForm";
+
+    return (
+
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialContents && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid={"RideForm-id"}
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialContents.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="day">Day</Form.Label>
+                <Form.Control
+                    data-testid={"RideForm-day"}
+                    id="day"
+                    type="text"
+                    isInvalid={Boolean(errors.day)}
+                    {...register("day", {
+                        required: "Day is required.",
+
```

Comment:
But if you don't make this a dropdown, at least use a validation pattern like this: (you may have to fix minor syntax errors):
```suggestion
     pattern: /^(Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|Sunday)$/i,
```

See also: https://www.react-hook-form.com/api/useform/register